### PR TITLE
[kddockwidgets] Add port

### PR DESCRIPTION
**Pull request by ThijsWithaar**
**Describe the pull request**

- #### What does your PR fix?  
  Adds a new port for [KDDockWidgets](https://www.kdab.com/development-resources/qt-tools/kddockwidgets)

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  Waiting for CI results. Since it's generic Qt code, expect it to be all.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes.

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by JonLiu1993** [comment]
@ThijsWithaar ,If pr is ready for review please let me know

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -147,6 +147,7 @@ cmcstl2:x86-windows        = skip
 coin:arm64-windows=fail
 coin:arm-uwp=fail
 coin:x64-uwp=fail
+colmap:x64-windows-static=fail
```
Comment (JonLiu1993) [inline scripts/ci.baseline.txt:150]: @ThijsWithaar ,Can you please clarify why this is added?

File: scripts/ci.baseline.txt [hunk 2]
```diff
@@ -497,6 +498,7 @@ jemalloc:x64-uwp=fail
 jemalloc:x64-windows-static=fail
 jinja2cpplight:arm-uwp=fail
 jinja2cpplight:x64-uwp=fail
+kddockwidgets:x64-windows-static=fail
```
Comment (JonLiu1993) [inline scripts/ci.baseline.txt:501]: Can you clarify why this is added?

File: ports/kddockwidgets/vcpkg.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "kddockwidgets",
+  "version-semver": "1.4.0",
+  "description": "KDAB's Dock Widget Framework for Qt",
+  "homepage": "https://www.kdab.com/development-resources/qt-tools/kddockwidgets",
+  "dependencies": [
+    "ecm",
+    "kf5coreaddons",
+    "kf5widgetsaddons",
+    "qt5-base",
```
Comment (JonLiu1993) [inline ports/kddockwidgets/vcpkg.json:10]: Could you please clarify why  kddockwidgets depend on them, I did not find anything related to these three ports in the source,

File: scripts/ci.baseline.txt (hunk 1 above)
Comment (ThijsWithaar) [reply scripts/ci.baseline.txt]: The CI build failed for x64-windows-static, with an error in colmap.
I was a bit at a loss a to why. Removed this line in a 2nd commit to reproduce the issue.

Comment (JonLiu1993) [reply scripts/ci.baseline.txt]: 'colmap' unrelated to this PR changes,please delete this line

**Comment by JonLiu1993** [comment]
@ThijsWithaar ,Could you please take a look?
https://dev.azure.com/vcpkg/public/_build/results?buildId=58065&view=artifacts&pathAsName=false&type=publishedArtifacts

**Comment by JonLiu1993** [comment]
@ThijsWithaar ,Could you please take a look?
````
CMake Error at /agent/_work/1/s/scripts/buildsystems/vcpkg.cmake:574 (_add_library):
  Target "kddockwidgets" links to target "Qt5::X11Extras" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  src/CMakeLists.txt:239 (add_library)

````

**Comment by JonLiu1993** [comment]
@ThijsWithaar ,Is this PR being worked on?

**Comment by ThijsWithaar** [comment]
> @ThijsWithaar ,Is this PR being worked on?

Effectively not, no. I thought this would be an easy one, but somehow I can't get the build-servers to pass the build. I'm sure that's me just not understanding things, but I am just stuck.

**Comment by JonLiu1993** [comment]
@ThijsWithaar ,Maybe the error related to Qt5:
````
CMake Warning at /agent/_work/1/s/scripts/buildsystems/vcpkg.cmake:786 (_find_package):
  By not providing "FindQt5X11Extras.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "Qt5X11Extras", but CMake did not find one.

  Could not find a package configuration file provided by "Qt5X11Extras" with
  any of the following names:

    Qt5X11ExtrasConfig.cmake
    qt5x11extras-config.cmake

  Add the installation prefix of "Qt5X11Extras" to CMAKE_PREFIX_PATH or set
  "Qt5X11Extras_DIR" to a directory containing one of the above files.  If
  "Qt5X11Extras" provides a separate development package or SDK, be sure it
  has been installed.
Call Stack (most recent call first):
  src/CMakeLists.txt:294 (find_package)


CMake Error at /agent/_work/1/s/scripts/buildsystems/vcpkg.cmake:576 (_add_library):
  Target "kddockwidgets" links to target "Qt5::X11Extras" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  src/CMakeLists.txt:239 (add_library)


CMake Warning:
  Manually-specified variables were not used by the project:

    VCPKG_PLATFORM_TOOLSET
    VCPKG_SET_CHARSET_FLAG
    _VCPKG_ROOT_DIR


CMake Generate step failed.  Build files cannot be regenerated correctly.

````

**Comment by JonLiu1993** [comment]
@ThijsWithaar, any progress?

**Comment by BillyONeal** [comment]
I'm closing this PR due to:
* Conflicts with master
* Contradicting https://github.com/microsoft/vcpkg/pull/21989
* Lack of response since September.

Sorry things took so long and hope any further experience you have with vcpkg is better :)

**Pull request closed without merging**